### DISCUSSION TITLE
Multi-icons

### DISCUSSION THREAD

**Discussion by kybos** [muse-sequencer/muse#1178] [Team Posts]
Hi,
I noticed this comment in icons.cpp:

```
// NOTICE: Although it is tempting to use multi-icons (addPixmap, addFile etc.),
//          certain styles do not support it, such as QtCurve.
//         Therefore the separate icons must be manually set upon each state.
```

On the other side, there are many icons that do exactly that:

```
      soloSVGIcon = new QIcon(":/svg/headphones_off.svg");
      soloSVGIcon->addFile(":/svg/headphones_on.svg", QSize(), QIcon::Normal, QIcon::On);
      // TODO
      soloProxySVGIcon = new QIcon(":/svg/headphones_off.svg");
      soloProxySVGIcon->addFile(":/svg/headphones_on.svg", QSize(), QIcon::Normal, QIcon::On);

      muteSVGIcon = new QIcon(":/svg/mute_off.svg");
      muteSVGIcon->addFile(":/svg/mute_on.svg", QSize(), QIcon::Normal, QIcon::On);

      trackEnableSVGIcon = new QIcon(":/svg/track_on.svg");
      trackEnableSVGIcon->addFile(":/svg/track_off.svg", QSize(), QIcon::Normal, QIcon::On);

      //recArmSVGIcon = new QIcon(":/svg/rec_arm_off_default_col.svg");
      recArmSVGIcon = new QIcon(":/svg/rec_arm_off.svg");
      recArmSVGIcon->addFile(":/svg/rec_arm_on.svg", QSize(), QIcon::Normal, QIcon::On);

      //recMasterSVGIcon = new QIcon(":/svg/rec_arm_off_default_col.svg");
      recMasterSVGIcon = new QIcon(":/svg/rec_arm_off.svg");
      recMasterSVGIcon->addFile(":/svg/rec_arm_on.svg", QSize(), QIcon::Normal, QIcon::On);
```
... and many more.

So which strategy is the right one?

**Comment by terminator356** [comment]
Broke my own rules, eh?
I wrote that comment about 8 years ago, because that was what I observed.
That is why I scrambled to make my own buttons such as seen in the mixer strips.
At least **those** buttons are guaranteed to show multiple icons no matter what style.

Gradually I began to forget and ignore that comment, in the name of progress and 
 simplicity of moving forward with standard buttons, where time is at such a premium.

So... I just installed QtCurve and checked:
Dummies! Still not working after all this time!

Forgive me and I hope you'll understand when I say:
Screw that, I'm not undoing all that work for one (or more?) styles.
Yet another stupid style to add to the "do not recommend" list.
I would much rather rattle the chains of the qtcurve team and get them to support that feature.

So we should change the comment from an absolute directive to a simpler warning,
that they "will not work with some styles".

Thanks for the reminder.
Tim.

**Comment by kybos** [comment]
I agree, we should not mess up the code because of incomplete or buggy styles.
One reason more to remove the style selection completely, as discussed earlier?
And to standardize the buttons in the mixer strip... they have given me headaches ;-). They don't look like buttons, technically they are not buttons (QFrame with custom paint method), so as a result there is no chance to customize/theme them in any way.
